pkg/config/config.go

```diff
@@ -466,19 +472,21 @@ func (c *MicroshiftConfig) ReadAndValidate(configFile string) error {
 	}
 	c.Cluster.DNS = clusterDNS
 
-	// If KAS advertise address is not configured then grab it from the service
+	// If KAS advertise address is not configured then compute it from the service
 	// CIDR automatically.
 	if len(c.KASAdvertiseAddress) == 0 {
 		// unchecked error because this was done when getting cluster DNS
 		_, svcNet, _ := net.ParseCIDR(c.Cluster.ServiceCIDR)
-		_, apiServerServiceIP, err := ctrl.ServiceIPRange(*svcNet)
-		if err != nil {
-			return fmt.Errorf("error getting apiserver IP: %v", err)
+		// Using first valid IP from next subnet as using the actual apiserver IP
```

Comment:
What it does is to get the next immediate subnet from the service CIDR and use that IP.
Given that we need a non-service-CIDR IP to setup for the apiserver endpoint this is the most trivial approach I could think of.
This is also why both parameters (the actual address and the lo interface getting it) are now configurable, since this is an additional IP and there might be cases where we need a different one because of collisions.

Reply:
Could you expand the comment in the code to explain all of that? The text that is there now might have been clear to someone who understood how we were already choosing an IP, but was not enough for me to understand what was going on.

Reply:
Done